uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/BuildForm/LocalisationPageFilter.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+class CRM_HRCore_Hook_BuildForm_LocalisationPageFilter {
+  /**
+   * Hides options on the localisation page
+   *
+   * @param string $formName
+   * @param CRM_Core_Form $form
+   */
+  public function handle($formName, &$form) {
+    if (!$this->shouldHandle($formName)) {
+      return;
+    }
+    
+    $this->filterLocalisationOptions($form);
+  }
+  
+  /**
+   * Checks if the hook should be handled.
+   *
+   * @param string $formName
+   *
+   * @return bool
+   */
+  private function shouldHandle($formName) {
+    if ($formName == CRM_Admin_Form_Setting_Localization::class) {
+      return TRUE;
+    }
+    
+    return FALSE;
+  }
+  
+  /**
+   * Hide options on localisation page if user does not have required permission
+   *
+   * @param CRM_Core_Form $form
+   */
+  private function filterLocalisationOptions($form) {
+    $canViewAllFields = CRM_Core_Permission::check('access root menu items and configurations ');
+    $settings = [
```

Comment:
I noticed that the CSS classes we want to hide are all the same format and match the names of these `$settings`. I haven't checked this totally, but if they are the same it would be nice to combine them, for example using a private property like:

```
private $restrictedSettings = [
  'customTranslateFunction',
  ...
];
```

and then use it in both `filterLocalisationOptions` and `hideOptionLabels`. 

This would prevent duplication.